crates/contracts/src/storage_accessible.rs

```diff
@@ -39,3 +45,38 @@ pub fn call(target: H160, code: Bytes, call: Bytes) -> CallRequest {
         ..Default::default()
     }
 }
+
+/// Simulates the specified `ethcontract::MethodBuilder` encoded as a
+/// `StorageAccessible` call.
+///
+/// # Panics
+///
+/// Panics if:
+/// - The method doesn't specify a target address or calldata
+/// - The function name doesn't exist or match the method signature
+/// - The contract does not have deployment code
+pub async fn simulate<T, R>(
+    web3: &Web3<T>,
+    contract: &ethcontract::Contract,
+    function_name: &str,
```

Comment:
Would be nice if MethodBuiler exposed function publicly to save this param.

Reply:
100% agree. Will try to push something to `ethcontract`. The other bit is that `to` address is being overwritten and would have allowed this to be even nicer!

Reply:
https://github.com/cowprotocol/ethcontract-rs/pull/963

Will fix the helper function once ☝️ merges and releases.